### PR TITLE
Add display SOC button

### PR DESCRIPTION
**Pull request by wangben718**
This PR try to add a button to show SOC column or hide it

### CODE REVIEW

**Review by LittleBeannie — APPROVED** [review]

**Comment by elong0527** [comment]
Please add an example in `vignettes` to illustrate the example and output.

File: R/format_ae_forestly.R [hunk 1]
```diff
@@ -224,6 +225,11 @@ format_ae_forestly <- function(
     name = reactable::colDef(
       header = "Adverse Events",
       minWidth = width_term, align = "right"
+    ),
+    soc_name = reactable::colDef(
+      header = "SOC Name",
+      minWidth = width_term, align = "right",
+      show = F
```
Comment (nanxstats) [inline R/format_ae_forestly.R:232]: Prefer `TRUE` and `FALSE` over `T` and `F`. https://style.tidyverse.org/syntax.html#logical-vectors

File: R/format_ae_forestly.R [hunk 2]
```diff
@@ -314,13 +320,28 @@ format_ae_forestly <- function(
...
+  # column hidden
+  hidden_item <- character()
+  for (item in seq_along(columns)) {
+    if(!"show" %in% names(columns[[item]])){
+      columns[[item]]$show <- TRUE
+    }
+    if (!columns[[item]]$show) {
+      hidden_item <- c(hidden_item, names(columns)[item])
+    }
+  }
```
Comment (nanxstats) [inline R/format_ae_forestly.R:332]: This for-loop should be vectorized to avoid growing the vector sequentially.

File: R/format_ae_forestly.R [hunk 3]
```diff
@@ -314,13 +320,28 @@ format_ae_forestly <- function(
...
+
+  hidden_item <- hidden_item[!hidden_item %in% "soc_name"]
+  # hidden_item <-  paste0("'", hidden_item, "'", collapse = ", ")
```
Comment (nanxstats) [inline R/format_ae_forestly.R:335]: Remove commented code

**Review by nanxstats — CHANGES_REQUESTED** [review]

**Comment by fukuhiro2023** [comment]
I have updated as below:

- Add argument `display_soc_toggle` in `ae_forestly()` to control the display of SOC toggle button.
- Fix formatting and vectorize the `for` loop based on the comments.
- Enable to show and hide the SOC column with a single button.

**Review by nanxstats — APPROVED** [review]

**Pull request merged**